packages/editor/src/components/blocks/text/Text.tsx

```diff
@@ -33,15 +39,27 @@ export const TextComponent: ComponentConfig<TextProps> = {
   fields: {
     content: { subsection: 'General', label: 'Content', type: 'textarea' },
     type: { subsection: 'General', label: 'Type', type: 'select', options: typeOptions },
-    icon: { subsection: 'General', label: 'Icon', type: 'text' },
+    icon: { subsection: 'Icon', label: 'Icon', type: 'text' },
+    iconStyle: { subsection: 'Icon', label: 'Icon style', type: 'select', options: iconOptions },
     ...baseComponentFields
   },
   quickActions: DEFAULT_QUICK_ACTIONS
 };
 
-const UiBlock = ({ content, icon }: UiComponentProps<TextProps>) => (
-  <p className='block-text'>
-    {icon && <IvyIcon icon={IvyIcons.InfoCircle} />}
-    {content}
-  </p>
-);
+const UiBlock = ({ content, icon, iconStyle }: UiComponentProps<TextProps>) => {
+  if (iconStyle == 'BLOCK') {
+    return (
+      <div className='text-icon-wrapper'>
+        <IvyIcon icon={IvyIcons.InfoCircle} />
+        <p className='block-text'>{content}</p>
+      </div>
+    );
+  } else {
+    return (
+      <p className='block-text'>
+        {icon && <IvyIcon icon={IvyIcons.InfoCircle} />}
+        {content}
+      </p>
+    );
+  }
```

Comment:
if you return something in an if, you don't have to write an else
```suggestion
  if (iconStyle == 'BLOCK') {
    return (
      <div className='text-icon-wrapper'>
        <IvyIcon icon={IvyIcons.InfoCircle} />
        <p className='block-text'>{content}</p>
      </div>
    );
  }
  return (
    <p className='block-text'>
      {icon && <IvyIcon icon={IvyIcons.InfoCircle} />}
      {content}
    </p>
  );
```